frontend/www/js/omegaup/arena/contest_practice.ts

```diff
@@ -114,32 +120,34 @@ OmegaUp.on('ready', () => {
       });
   }
 
-  // The hash is of the form `#problems/${alias}`.
-  const problemMatch = /#problems\/([^/]+)/.exec(window.location.hash);
-  const problemAlias = problemMatch?.[1] ?? null;
-  if (problemAlias && problemMatch) {
-    // This needs to be set here and not at the top because it depends
-    // on the `navigate-to-problem` callback being invoked, and that is
-    // not the case if this is set a priori.
-    contestPractice.problem = { alias: problemAlias, runs: [] };
-    const popupDisplayed = problemMatch.input.split('/')[2];
-    if (popupDisplayed) {
-      if (popupDisplayed === 'new-run') {
-        contestPractice.popupDisplayed = PopupDisplayed.RunSubmit;
-      }
-      if (popupDisplayed.includes('show-run')) {
-        contestPractice.popupDisplayed = PopupDisplayed.RunDetails;
-      }
-    }
-  }
-  // The hash is of the form `#clarifications/[new]`.
-  const clarificationMatch = /#clarifications\/([^/]+)/.exec(
+  // The hash is of the forms:
+  // - `#problems/${alias}`
+  // - `#problems/${alias}/new-run`
+  // - `#problems/${alias}/show-run:xyz`
+  // - `#clarifications/${alias}/new`
+  // and all the matching forms in the following regex
+  const match = /#(?<tab>\w+)\/(?<problem>[^/]+)\/?(?<popup>[^/]*)/g.exec(
```

Comment:
```suggestion
  const match = /#(?<tab>\w+)\/(?<alias>[^/]+)(?:\/(?<popup>[^/]+))?/g.exec(
```

es mejor ser consistente con el uso de `alias` (o también se vale usar `${problem}` en L124-L127).

también, se puede usar `(?:...)` para denotar un [non-capturing group](https://www.regular-expressions.info/brackets.html) para agrupar cosas (en este caso hacerlas opcionales como grupo) SIN que interfieran con los grupos de captura. así también evitamos el uso de `*` cuando podemos usar el `+` que es mejor.

Reply:
Ah, cierto. Lo cambio.